build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts

```diff
@@ -0,0 +1,157 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fs from "node:fs/promises";
+import JSON5 from "json5";
+
+import type { Logger } from "@fluidframework/build-tools";
+import * as resolve from "resolve.exports";
+import type { PackageJson, TsConfigJson } from "type-fest";
+import { BaseCommand, Node10CompatExportData } from "../../library/index.js";
+import type { CommandLogger } from "../../logging.js";
+
+export default class GenerateNode10EntrypointsCommand extends BaseCommand<
+	typeof GenerateNode10EntrypointsCommand
+> {
+	static readonly description =
+		`Generates node10 type declaration entrypoints for Fluid Framework API levels (/alpha, /beta, /internal etc.) as found in package.json "exports"`;
+
+	public async run(): Promise<void> {
+		const packageJson = await readPackageJson();
+
+		const tsconfig = await readTsConfig();
+
+		let emitDeclarationOnly = false;
+		if (tsconfig.compilerOptions?.emitDeclarationOnly !== undefined) {
+			emitDeclarationOnly = true;
+		}
+
+		const mapNode10CompatExportPathToData = mapExportPathsFromPackage(
+			packageJson,
+			emitDeclarationOnly,
+			this.logger,
+		);
+
+		if (mapNode10CompatExportPathToData.size === 0) {
+			throw new Error(
+				'There are no API level "exports" requiring Node10 type compatibility generation.',
+			);
+		}
+
+		await generateNode10TypeEntrypoints(mapNode10CompatExportPathToData, this.logger);
+	}
+}
+
+export function mapExportPathsFromPackage(
+	packageJson: PackageJson,
+	emitDeclarationOnly: boolean,
+	logger?: Logger,
+): Map<string, Node10CompatExportData> {
```

Comment:
Should be replaceable with a shared function, maybe even the one in generate:typetests.